### PR TITLE
Feat/latency layer2 config

### PR DESCRIPTION
**Pull request by shreyvishal**

### CODE REVIEW

File: src/middlewares/schemaValidator.middleware.ts [hunk 1]
```diff
@@ -5,8 +5,54 @@ import { Locals } from "../interfaces/locals.interface";
...
+      apiSpec,
+      validateRequests: true,
+      validateResponses: false,
+      $refParser: {
+        mode: "dereference"
+      }
+    });
+    cachedSpecFile[specFile] = specFile;
+  }
+  return cachedOpenApiValidator;
```
Comment (vbabuEM) [inline src/middlewares/schemaValidator.middleware.ts:54]: Shouldnt this return cachedOpenApiValidator[specFile]

File: src/middlewares/schemaValidator.middleware.ts [hunk 2]
```diff
@@ -5,8 +5,54 @@ import { Locals } from "../interfaces/locals.interface";
...
+import logger from "../utils/logger.utils";
+
+// Cache object
+const apiSpecCache: { [filename: string]: OpenAPIV3.Document } = {};
+
+let cachedOpenApiValidator: {
+  [filename: string]: express.RequestHandler[];
+} = {};
+
+let cachedSpecFile: { [filename: string]: string } = {};
```
Comment (vbabuEM) [inline src/middlewares/schemaValidator.middleware.ts:21]: This is probably not required.

File: src/middlewares/schemaValidator.middleware.ts [hunk 3]
```diff
@@ -85,5 +138,5 @@ export const openApiValidatorMiddleware = async (
     };
     walkStack(0);
   };
-  walkSubstack([...openApiValidator], req, res, next);
+  walkSubstack([...openApiValidator[specFile]], req, res, next);
```
[the remark this answers is not in the document]
Comment (shreyvishal) [reply src/middlewares/schemaValidator.middleware.ts]: resolved this

File: src/middlewares/schemaValidator.middleware.ts (hunk 1 above)
Comment (shreyvishal) [reply src/middlewares/schemaValidator.middleware.ts]: resolved this

**Pull request merged**